addons/website/static/src/snippets/s_dynamic_filter_carousel/000.js

```diff
@@ -0,0 +1,122 @@
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.uniqueId = _.uniqueId('s_dynamic_filter_carousel_');
+        this._onResizeChange = _.debounce(this._refresh_carousel_content, 100);
+    },
+    /**
+     * @override
+     */
+    start: function () {
+        return this._super.apply(this, arguments)
+            .then(() => {
+                $(window).resize(() => {
+                    this._onResizeChange();
+                });
+                this._refresh_carousel_content();
+            });
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        this._super(...arguments);
+        this.$el.addClass('d-none');
+        this.$el.find('.carousel-inner').html('');
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _create_dataGroups: function (data) {
+        const number_of_records_per_chunk = this._number_of_records_per_chunk();
+        const data_groups = Array(Math.ceil(data.length / number_of_records_per_chunk))
+            .fill()
+            .map((_, index) => index * number_of_records_per_chunk)
+            .map(begin => data.slice(begin, begin + number_of_records_per_chunk));
+        for (let i = 0; i < data_groups.length; i++) {
+            data_groups[i] = data_groups[i].map((data) => {
+                // Wrap data into carousel cards and adapt layout according to number of records per chunk.
+                data = $('<div></div>')
+                        .addClass('col-md-' + Math.floor(12/number_of_records_per_chunk))
+                        .append($('<div class="card h-100"></div>')
+                        .append($(data)))
+                        .get(0)
+                        .outerHTML;
+                return data;
+            });
+        }
+        return data_groups;
+
+    },
+    /**
+     * @private
+     */
+    _fetch: function () {
+        return this._rpc({
+            'route': '/website/snippet/dynamic_filter',
+            'params': {
+                'filter_id': (this.$target[0].dataset.filterId && parseInt(this.$target[0].dataset.filterId)) || 0,
+                'template_id': (this.$target[0].dataset.templateId && parseInt(this.$target[0].dataset.templateId)) || 0,
+                'call_to_action': this.$target[0].dataset.callToAction === 'true',
+                'number_of_records_per_chunk': this._number_of_records_per_chunk
+            },
+        });
+    },
+    /**
+     * @private
+     */
+    _number_of_records_per_chunk: function() {
+        return config.device.size_class <= config.device.SIZES.SM ? 1 : 4;
+    },
+    /**
+     * @private
+     */
+    _refresh_carousel_content: function () {
+        return this._fetch()
+            .then(data => this._create_dataGroups(data))
```

Comment:
while, resize need to re-render the caroussel in 1 by row on small screen, I don't understand why we need to refetch all the records each time, e already got the information.